### ISSUE TITLE
Plan D: K-1 pass-through + Sch E Part II + 8995 + 8582 + 1099-G

### ISSUE THREAD

**Issue by translunar** [translunar/tenforty#17]
# Plan D: K-1 + Sch E Part II + 8995 + 8582 + 1099-G

## Scope

Add support for pass-through income (K-1 from S-corp / partnership / estate-trust), the forms that surround it (Sch E Part II, 8995 QBI, 8582 passive loss), and 1099-G income (unemployment, state refund recovery, other boxes).

Closes follow-ups filed from Plan C e2e failures:
- #15 (partial — Sch D direct capital gains from K-1 boxes 7/8, not yet full 1099-B)
- #16 (full — K-1 / Sch E Part II pass-through)

## In scope

- **K-1** for all three entity types: S-corp (1120-S), Partnership (1065), Estate/Trust (1041). Unified \`ScheduleK1\` dataclass, entity_type enum discriminates.
- **Sch E Part II**: up to 4 K-1 rows (A/B/C/D), mix of types allowed.
- **Form 8995 (simple QBI)**: 20% QBI deduction for taxpayers below the single \per Rev. Proc. 2024-40 (approx $197,300 single, $394,600 MFJ) / MFJ \See Rev. Proc. 2024-40 for authoritative 2025 figures threshold. Flows to 1040 line 13.
- **Form 8582 (current-year passive activity loss)**: determines allowed vs suspended passive loss. \$25k special allowance for rental with active participation (phased out \$100k-\$150k AGI). Prior-year carryforward as scenario input.
- **1099-G (all 7 boxes)**: unemployment (box 1), state tax refund (box 2, tax-benefit-rule applied), RTAA (5), taxable grants (6), ag payments (7), market gain (9), federal withholding (box 4).
- K-1 separately-stated items fan out to Sch B (interest, dividends), Sch D (cap gains), Sch E Part II (ordinary biz + rental + royalties), Form 8995 (QBI aggregate).
- Flattener \`_reject_unhandled\` gates for K-1 and 1099-G are removed; feature gates replaced with explicit attestations.

## Out of scope (attestations)

Each of these requires a dedicated attestation field on ScenarioConfig. None set → \`ValueError\` at scenario load. \`False\` + trigger fired → \`NotImplementedError\` at compute. \`True\` → proceed.

- \`acknowledges_qbi_below_threshold: bool | None\` — triggered when QBI present; if taxable income exceeds threshold, \`False\` raises (Form 8995-A not implemented).
- \`acknowledges_unlimited_at_risk: bool | None\` — triggered when K-1 present; Form 6198 scoped out.
- \`basis_tracked_externally: bool | None\` — triggered when K-1 present; basis worksheets scoped out.
- \`acknowledges_no_partnership_se_earnings: bool | None\` — partnership K-1 with non-zero box 14 raises; Schedule SE out of scope.
- \`acknowledges_no_section_1231_gain: bool | None\` — K-1 section 1231 gain scoped out (Form 4797 not implemented).
- \`acknowledges_no_more_than_four_k1s: bool | None\` — Sch E Part II continuation not supported; >4 K-1s raises.
- \`acknowledges_no_k1_credits: bool | None\` — K-1 box 13/15 credits scoped out.
- \`acknowledges_no_section_179: bool | None\` — K-1 section 179 scoped out.

## Architecture

Same pattern as Plans B/C: native Python compute + oracle cross-check + \`PdfFiller\` emission. Reuses all infrastructure from Plans A/B/C.

### New files

- \`tenforty/forms/sch_e_part_ii.py\` — K-1 compute, produces Sch E Part II rows + fan-out to Sch B / Sch D / Form 8995
- \`tenforty/forms/f8995.py\` — QBI deduction compute (simple only)
- \`tenforty/forms/f8582.py\` — Current-year passive loss compute + per-activity carryforward output
- \`tenforty/mappings/pdf_sch_e.py\` — extend Plan B's Part I module with Part II mapping
- \`tenforty/mappings/pdf_f8995.py\`
- \`tenforty/mappings/pdf_f8582.py\`
- \`pdfs/federal/2025/f8995.pdf\` and \`f8582.pdf\` — download if not present

### Scenario model additions

\`\`\`python
@dataclass
class ScheduleK1:
    entity_name: str
    entity_ein: str
    entity_type: Literal['s_corp', 'partnership', 'estate_trust']
    material_participation: bool
    ordinary_business_income: float = 0.0
    net_rental_real_estate: float = 0.0
    other_net_rental: float = 0.0
    interest_income: float = 0.0
    ordinary_dividends: float = 0.0
    qualified_dividends: float = 0.0
    royalties: float = 0.0
    net_short_term_capital_gain: float = 0.0
    net_long_term_capital_gain: float = 0.0
    other_income: float = 0.0
    qbi_amount: float = 0.0
    prior_year_passive_loss_carryforward: float = 0.0
    # Scope-out fields (raise if non-zero):
    section_1231_gain: float = 0.0
    section_179_deduction: float = 0.0
    partnership_self_employment_earnings: float = 0.0

@dataclass
class Form1099G:
    payer: str
    unemployment_compensation: float = 0.0
    state_tax_refund: float = 0.0
    state_tax_refund_tax_year: int | None = None
    federal_tax_withheld: float = 0.0
    rtaa_payments: float = 0.0
    taxable_grants: float = 0.0
    agriculture_payments: float = 0.0
    market_gain: float = 0.0
\`\`\`

Plus \`scenario.schedule_k1s: list[ScheduleK1]\` and \`scenario.form1099_g: list[Form1099G]\`.

### Compute flow per K-1

Fan-out rules (same across entity types; tenforty normalizes per-type K-1 box numbers into the dataclass fields above):

| Scenario field | Flows to |
|---|---|
| ordinary_business_income, net_rental_*, other_income, royalties | Sch E Part II row (active/passive column determined by material_participation) |
| interest_income | Sch B line 1, payer = entity_name |
| ordinary_dividends | Sch B line 5; qualified → 1040 line 3a |
| net_short_term_capital_gain | Sch D line 5 |
| net_long_term_capital_gain | Sch D line 12 |
| qbi_amount | Form 8995 aggregate |
| prior_year_passive_loss_carryforward | Form 8582 Part I |

### 8582 compute (when passive activities present)

1. Identify passive activities: K-1s with material_participation=False; Sch E Part I rentals.
2. Sum passive income across sources.
3. Sum passive losses.
4. Special allowance: \`max(0, 25000 - 0.5 * max(0, MAGI - 100000))\`; \$0 at MAGI ≥ \$150k.
5. Allowed loss = min(passive income + special allowance, total passive loss).
6. Per-activity suspended = this year's loss × (1 - allowed/total).
7. Output: each passive activity's suspended loss as carryforward for next year's scenario.

### 8995 compute (simple QBI)

\`qbi_deduction = min(0.20 * qbi_total, 0.20 * (taxable_income_before_qbi - net_capital_gains))\`.

If taxable income exceeds threshold, raise (attest_qbi_below_threshold gate fires).

### 1099-G fanout

| Box | Flows to |
|---|---|
| 1 Unemployment | Sch 1 line 7 |
| 2 State refund | Sch 1 line 1 (tax benefit rule: taxable only if itemized prior year; scenario provides \`prior_year_itemized: bool\`) |
| 4 Federal withholding | 1040 line 25b |
| 5, 6, 7, 9 (RTAA / grants / ag / market gain) | Sch 1 line 8z (other income) |

## Testing strategy

- \`unittest.TestCase\` throughout.
- **XLSX oracle** for 1040-level: Form 8995 line 11 QBI deduction, Form 8582 line 11 allowed loss, 1099-G→Sch 1 totals.
- **Independent IRS-instructions reference** (hand-coded, in \`tests/oracles/\`) for per-K-1 fan-out. Lives on \`oracle/k1-reference\` branch; built by a different agent than the production compute; not visible to planner or implementer.
- 3 of 11 pre-existing Plan A baseline e2e failures flip green once K-1 support lands (TestE2EScheduleK1 class).
- 8 attestations each get 3-way test coverage (None/False-triggered/True-proceed).

## Task order (~17 tasks)

1. Scenario model: ScheduleK1, Form1099G, 8 new attestations
2. Flattener: K-1 dispatch per entity_type, 1099-G dispatch; remove related \`_reject_unhandled\` gates
3. \`forms/sch_e_part_ii.py\` native compute
4. Sch E Part II cross-check (against IRS-instructions reference) + \`_should_emit\` predicate
5. \`mappings/pdf_sch_e.py\` Part II mapping extension + wire
6. \`forms/f8995.py\` QBI compute (simple)
7. f8995 XLSX oracle cross-check + \`_should_emit\`
8. \`mappings/pdf_f8995.py\` + wire
9. \`forms/f8582.py\` PAL compute + carryforward output
10. f8582 XLSX oracle cross-check + \`_should_emit\`
11. \`mappings/pdf_f8582.py\` + wire
12. 1099-G compute in \`forms/sch_1.py\` extension
13. 1099-G box 4 withholding flow to 1040 line 25b
14. State-refund tax-benefit-rule compute (scenario prior_year_itemized flag)
15. Per-entity-type K-1 fan-out tests (cross-check against \`tests/oracles/\` reference)
16. End-to-end smoke scenarios (k1_scorp_basic, k1_partnership_passive, k1_estate_trust, k1_mixed_four, unemployment_withholding, state_refund_benefit_rule)
17. Final behavior-preservation check + PR

## Prerequisites

- Plan C merged to main (done; \`eee7bb6\`).
- \`tests/oracles/k1_reference.py\` committed on \`oracle/k1-reference\` branch by a separate agent (tenforty-ca, research-only role — no visibility to the planner or implementer of this plan).

## Iron-law notes

- All new test files use \`unittest.TestCase\`.
- No PII in the repo or issues — fixture scenarios use round synthetic amounts and placeholder names.
- Every scope-out is an active attestation gate; no silent skips.
- Imports at top of file; no inline imports.

**Comment by translunar** [comment]
Plan D moved to sub-issue: https://github.com/translunar/tenforty/issues/18

**Comment by translunar** [comment]
Closed by #19 (Plan D merged to main as 576c4ee).
